### DISCUSSION TITLE
Lesson 7 : testWithDrawWithASingleFunder function with Gas Price doesn't charges gas.

### DISCUSSION THREAD

**Discussion by Tanmay53** [Cyfrin/foundry-full-course-cu#542] [Q&A] [answered]
I am running the **FundMeTest.t.sol** from Lesson 7.
In the function named **testWithDrawWithASingleFunder**, I tried it with gas price on Anvil (local network).

Though, as per the *gasUsed* calculation, I get a gas price, but when I subtract the owner's starting balance and the FundeMe balance from the owner's ending balance, I get a zero which I think should be equal to the *gasUsed*.

Asseting : gasUsed  = endingOwnerBalance - startingOwnerBalance - startingFundMeBalance 

```solidity
    function testWithdrawWithSingleFunder() public funded {
        uint startingOwnerBalance = fundMe.i_owner().balance;
        uint startingFundMeBalance = address(fundMe).balance;

        uint gasStart = gasleft();

        vm.txGasPrice(GAS_PRICE);
        vm.prank(fundMe.i_owner());
        fundMe.withdraw();

        uint gasEnd = gasleft();
        uint gasUsed = (gasStart - gasEnd) * tx.gasprice;

        uint endingOwnerBalance = fundMe.i_owner().balance;
        uint endingFundMeBalance = address(fundMe).balance;

        console.log(gasUsed); // 10667
        console.log(endingOwnerBalance - startingOwnerBalance - startingFundMeBalance); // 0

        // assertEq(endingFundMeBalance, 0);
        // assertEq(startingOwnerBalance + startingFundMeBalance, endingOwnerBalance);
        assertEq(gasUsed, endingOwnerBalance - startingOwnerBalance - startingFundMeBalance);
    }
```

**Comment by Naman1729** [comment, marked as answer]
Hey @Tanmay53 , Gas amount is taken after the function is fully executed . Here, testWithdrawWithSingleFunder() is a calling withdraw() function which complete its task . but , Gas fee is not taken yet after the testWithdrawWithSingleFunder() is complete it whole transactions then gas fee is added . So , that why this assert is not working .
